src/AppBundle/Twig/AppExtension.php

```diff
@@ -454,6 +455,9 @@ public function replag()
                 $project = 'enwiki';
             }
 
+            $projectName = ProjectRepository::getProject($project, $this->container)
+                ->getDatabaseName();
+
             $stmt = "SELECT lag FROM `heartbeat_p`.`heartbeat` h
             RIGHT JOIN `meta_p`.`wiki` w ON concat(h.shard, \".labsdb\")=w.slice
             WHERE dbname LIKE :project OR name LIKE :project OR url LIKE :project LIMIT 1";
```

Comment:
Does this mean we only need to check against `dbname` here?

Reply:
That's a good point... we could do the same as we do [`ProjectRepository::getOne`](https://github.com/x-tools/xtools/blob/master/src/Xtools/ProjectRepository.php#L163-L178), but then we're duplicating code. Also above we're re-querying for the project db name, when it's probably already there (at least on result pages, where there's always a `project` in Twig).

If we want to show the replag on index pages in addition to result pages, maybe we should do it like [AutoEditsController](https://github.com/x-tools/xtools/blob/master/src/AppBundle/Controller/AutomatedEditsController.php#L97) where we pass in `project` to Twig, so there will always be a project object (except on the Home and About pages). That way in `base.html.twig` we could call `{% set replagValue = project is defined ? replag(project.dbName) : 0 %}`; or even better, make a `Project::getReplag` function so we only have to put:
```
{% if project is defined and project.replag > getReplagThreshold() %}
    <p class="alert alert-warning xt-alert">
    ...
{% endif %}
```
and we don't need the AppExtension helper.

Reply:
I like @MusikAnimal 's solution as a long-term solution.  For now, this is a reasonable short-term solution.